src/terminal/terminal.mli

```diff
@@ -3,102 +3,24 @@
    Distributed under the MIT license. See terms at the end of this file.
   ————————————————————————————————————————————————————————————————————————————*)
 
-(** This library provides a small set of standard utility functions for
-    interacting with terminals. *)
-
-module Color : sig
-  type t
-  (** The type of colours that can be rendered to a terminal. *)
-
-  (** {1 4-bit ANSI colours}
-
-      Colours built using {!ansi} will be rendered using the standard
-      {{:https://en.wikipedia.org/wiki/ANSI_escape_code#3-bit_and_4-bit} 4-bit
-        ANSI escape codes} for terminals. The actual colours displayed to the
-      user depend on their terminal configuration / theme, ensuring that they
-      look natural in context. *)
-
-  type plain =
-    [ `black | `blue | `cyan | `green | `magenta | `red | `white | `yellow ]
-
-  val ansi : [ plain | `bright of plain ] -> t
-
-  (** {1 24-bit RGB colours}
-
-      Most modern terminals offer support for full 24-bit RGB colour (called
-      "{{:https://en.wikipedia.org/wiki/ANSI_escape_code#24-bit} true colour}")
-      in addition to the 16 original ANSI colours. These colours are rendered
-      exactly as requested, offering greater flexibility at the risk of clashing
-      with the user's theming. *)
-
-  val rgb : int -> int -> int -> t
-  (** [rgb r g b] is the RGB24 colour with the given red, green and blue colour
-      components respectively. Raises [Invalid_argument] if any of the
-      components are outside the range [\[0, 255\]]. *)
-
-  val hex : string -> t
-  (** [hex s] is the RGB24 colour given by the
-      {{:https://en.wikipedia.org/wiki/Web_colors#Hex_triplet} hex triplet} [s],
-      which must start with [#]. Examples:
-
-      - [hex "#FF8C00"] = [rgb 0xFF 0x8C 0x00]
-      - [hex "#fa0"] = [rgb 0xFF 0xAA 0x00]
-
-      Raises [Invalid_argument] if the given string is not a [#]-prefixed hex
-      triplet. *)
-
-  val pp_dump : Format.formatter -> t -> unit
-  (** Pretty-print a colour with an unspecified format. *)
+(** @inline *)
+include
+  module type of Terminal_ansi
+    with type Color.t = Terminal_ansi.Color.t
+     and type Style.t = Terminal_ansi.Style.t
```

Comment:
This is a bit ugly, but we have a diamond dependency:

```
terminal.ansi <------- terminal
 ^                       ^
 |                       |
progress.engine <----- progress
```

and `progress` needs to see that `terminal` and `progress.engine` are using the same color/style types.